src/ScalarMessaging.js

```diff
@@ -218,6 +223,18 @@ function returnStateEvent(event, roomId, eventType, stateKey) {
     sendResponse(event, stateEvent.getContent());
 }
 
+var currentRoomId = null;
+
+// Listen for when a room is viewed
+dis.register(onAction);
+function onAction(payload) {
+    switch (payload.action) {
+        case 'view_room':
+            currentRoomId = payload.room_id;
+        break;
```

Comment:
I would just do:
```javascript
function onAction(payload) {
  if (payload.action !== "view_room") {
    return;
  }
  currentRoomId = payload.room_id;
}
```

We don't expect to add more listeners so the `switch` is a bit premature.